### DISCUSSION TITLE
Capturing zone status with PC5108 expander and an Arduino

### DISCUSSION THREAD

**Discussion by anistor** [taligentx/dscKeybusInterface#206] [dscKeybusInterface Discussion]
Hi, 
this is a wonderful project that allowed me to understand a few pieces of the DSC keybus protocol, but I'm very new here and I have a few questions, if somebody is able to shed some light.

My goal is to use several PC5108 modules (8 zone expander) with an Arduino Mega, without a central panel and keypad setup. Just the zone expanders, with the 4 wire bus and the arduino.  In a regular setup there is an alarm panel driving the bus and the arduino can listen to it using this library, and also write commands to the bus occasionally. But In my setup the arduino should  drive the bus by itself, polling the zone expanders. I need to emulate the sequence of messages that would normally happen on the bus when the panel queries the expanders, so the PC5108 modules will reply appropriately. Also would be great if the modules would send zone change notifications instead of being polled constantly (not sure if this is how it normally works). Can somebody explain or point me to the relevant source code or documentation that would allow me to understand how is the central alarm panel reading the zones conencted to these expanders? Thanks a lot!

PS: Please note my question is just about how the bus works, I'm not asking for assistance for writing the code or the electrical interfacing required for driving the bus :)

**Comment by anistor** [comment]
It seems @taligentx is already trying to run a setup without a panel here https://github.com/taligentx/dscKeybusInterface/issues/200 and @Dilbert66 is also tinkering with zone expanders here https://github.com/taligentx/dscKeybusInterface/issues/64. Could you guys provide any pointers please to my question?

**Comment by Dilbert66** [comment]
The discussion in https://github.com/taligentx/dscKeybusInterface/issues/64  includes all relevent protocol packets and the emulation code implemented in branch "expander" of this repository should answer all your questions.  Also , Nikhil has the protocol well documented in the dev branch .  I suspect that Nikhil  might also implement this in his panel emulator code as it should be fairly easy to add now since he has the bulk of the main communications loop working
